File: app/src/main/java/org/stepic/droid/store/dao/DownloadEntityDaoImpl.java
package org.stepic.droid.store.dao;

import android.content.ContentValues;
import android.database.Cursor;
import android.database.sqlite.SQLiteOpenHelper;

import org.stepic.droid.model.DownloadEntity;
import org.stepic.droid.store.structure.DbStructureSharedDownloads;

public class DownloadEntityDaoImpl extends DaoBase<DownloadEntity> {
    public DownloadEntityDaoImpl(SQLiteOpenHelper openHelper) {
        super(openHelper);
    }

    @Override
    public DownloadEntity parsePersistentObject(Cursor cursor) {
        DownloadEntity downloadEntity = new DownloadEntity();

        int indexDownloadId = cursor.getColumnIndex(DbStructureSharedDownloads.Column.DOWNLOAD_ID);
        int indexStepId = cursor.getColumnIndex(DbStructureSharedDownloads.Column.STEP_ID);
        int indexVideoId = cursor.getColumnIndex(DbStructureSharedDownloads.Column.VIDEO_ID);
        int indexThumbnail = cursor.getColumnIndex(DbStructureSharedDownloads.Column.THUMBNAIL);
        int indexQuality = cursor.getColumnIndex(DbStructureSharedDownloads.Column.QUALITY);

        downloadEntity.setDownloadId(cursor.getLong(indexDownloadId));
        downloadEntity.setStepId(cursor.getLong(indexStepId));
        downloadEntity.setVideoId(cursor.getLong(indexVideoId));
        downloadEntity.setThumbnail(cursor.getString(indexThumbnail));
        downloadEntity.setQuality(cursor.getString(indexQuality));

        return downloadEntity;
    }

    @Override
    public String getDbName() {
        return DbStructureSharedDownloads.SHARED_DOWNLOADS;
    }

    @Override
    public ContentValues getContentValues(DownloadEntity downloadEntity) {
        ContentValues values = new ContentValues();

        values.put(DbStructureSharedDownloads.Column.DOWNLOAD_ID, downloadEntity.getDownloadId());
        values.put(DbStructureSharedDownloads.Column.VIDEO_ID, downloadEntity.getVideoId());
        values.put(DbStructureSharedDownloads.Column.STEP_ID, downloadEntity.getStepId());
        values.put(DbStructureSharedDownloads.Column.THUMBNAIL, downloadEntity.getThumbnail());
        values.put(DbStructureSharedDownloads.Column.QUALITY, downloadEntity.getQuality());

        return values;
    }

    @Override
    public String getDefaultPrimaryColumn() {
        return DbStructureSharedDownloads.Column.DOWNLOAD_ID;
    }

    @Override
    public String getDefaultPrimaryValue(DownloadEntity persistentObject) {
        return persistentObject.getDownloadId() + "";
    }
}

File: app/src/main/java/org/stepic/droid/storage/dao/DownloadEntityDaoImpl.kt
package org.stepic.droid.storage.dao

import android.content.ContentValues
import android.database.Cursor
import android.database.sqlite.SQLiteDatabase

import org.stepic.droid.model.DownloadEntity
import org.stepic.droid.storage.structure.DbStructureSharedDownloads
import javax.inject.Inject

class DownloadEntityDaoImpl @Inject constructor(openHelper: SQLiteDatabase) : DaoBase<DownloadEntity>(openHelper) {

    public override fun parsePersistentObject(cursor: Cursor): DownloadEntity {
        val indexDownloadId = cursor.getColumnIndex(DbStructureSharedDownloads.Column.DOWNLOAD_ID)
        val indexStepId = cursor.getColumnIndex(DbStructureSharedDownloads.Column.STEP_ID)
        val indexVideoId = cursor.getColumnIndex(DbStructureSharedDownloads.Column.VIDEO_ID)
        val indexThumbnail = cursor.getColumnIndex(DbStructureSharedDownloads.Column.THUMBNAIL)
        val indexQuality = cursor.getColumnIndex(DbStructureSharedDownloads.Column.QUALITY)

        return DownloadEntity(
                downloadId = cursor.getLong(indexDownloadId),
                stepId = cursor.getLong(indexStepId),
                videoId = cursor.getLong(indexVideoId),
                thumbnail = cursor.getString(indexThumbnail),
                quality = cursor.getString(indexQuality)
        )
    }

    public override fun getDbName(): String {
        return DbStructureSharedDownloads.SHARED_DOWNLOADS
    }

    public override fun getContentValues(downloadEntity: DownloadEntity): ContentValues {
        val values = ContentValues()

        values.put(DbStructureSharedDownloads.Column.DOWNLOAD_ID, downloadEntity.downloadId)
        values.put(DbStructureSharedDownloads.Column.VIDEO_ID, downloadEntity.videoId)
        values.put(DbStructureSharedDownloads.Column.STEP_ID, downloadEntity.stepId)
        values.put(DbStructureSharedDownloads.Column.THUMBNAIL, downloadEntity.thumbnail)
        values.put(DbStructureSharedDownloads.Column.QUALITY, downloadEntity.quality)

        return values
    }

    public override fun getDefaultPrimaryColumn(): String {
        return DbStructureSharedDownloads.Column.DOWNLOAD_ID
    }

    public override fun getDefaultPrimaryValue(persistentObject: DownloadEntity): String {
        return persistentObject.downloadId.toString() + ""
    }
}
